sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -256,7 +256,10 @@ object Analyzer {
     "spark.sql.expressionTreeChangeLog.level"
   )
 
-  def retainResolutionConfigsForAnalysis(newConf: SQLConf, existingConf: SQLConf): Unit = {
+  def retainResolutionConfigsForAnalysis(
+      newConf: SQLConf,
+      existingConf: SQLConf,
+      createSparkVersion: Option[String] = None): Unit = {
```

Comment:
```suggestion
      createSparkVersion: String = ""): Unit = {
```

Reply:
empty string means unknown spark version, which is also the default value of `CatalogTable#createVersion`